core/src/main/java/com/linecorp/armeria/client/limit/AbstractConcurrencyLimitingClient.java

```diff
@@ -54,17 +56,14 @@
 public abstract class AbstractConcurrencyLimitingClient<I extends Request, O extends Response>
         extends SimpleDecoratingClient<I, O> {
 
-    private static final long DEFAULT_TIMEOUT_MILLIS = 10000L;
-
-    private final int maxConcurrency;
-    private final long timeoutMillis;
-    private final AtomicInteger numActiveRequests = new AtomicInteger();
     private final Queue<PendingTask> pendingRequests = new ConcurrentLinkedQueue<>();
+    private final ConcurrencyLimit concurrencyLimit;
+    private final AtomicInteger numActiveRequests;
```

Comment:
How about replacing this field with a getter method like this following? I see no reason to keep it in a separate field.

```java
private AtomicInteger numActiveRequests() {
    return concurrencyLimit.numActiveRequests();
}
```

Reply:
I left this one out, as the value is used throughout the class it will make it verbose